dashboard/graphql/objects.py

```diff
@@ -142,7 +143,14 @@ def resolve_median_grade(parent, info):
         return info.context.submissions_by_assignment_id_loader.load(parent.id).then(
             lambda submissions: AssignmentType._median_grade_lambda(parent, info, submissions)
         )
-
+    
+    def resolve_due_date(parent, info):
+        original_due_date = parent.due_date
+        if original_due_date is None:
+            return None
+        new_timezone = pytz.timezone(info.context.session.get('time_zone'))
+        return original_due_date.astimezone(new_timezone)
```

Comment:
We expect `session['time_zone']` to be set here, but there may be some rare case in which it is not.  If it's not, the call to `session.get('time_zone')` could return `None`, which will cause `pytz.timesone()` to give an error.

We could avoid the error here by returning the default MyLA time zone if it's not set in the session.  Add that to the call like this… `session.get('time_zone', default_time_zone)`.  I haven't looked to see whether the default time zone is available in this part of the code.  It should be added if it's not.

Reply:
good point, added the default time zone to that script for this case.